python/cuml/decomposition/pca.pyx

```diff
@@ -589,20 +603,23 @@ class PCA:
         cdef uintptr_t singular_vals_ptr = self.singular_values_ptr
         cdef uintptr_t mean_ptr = self.mean_ptr
 
+        cdef cumlHandle* h_ = <cumlHandle*><size_t>self.handle.getHandle()
         if gdf_datatype.type == np.float32:
-            pcaTransform(<float*> input_ptr,
-                               <float*> components_ptr,
-                               <float*> trans_input_ptr,
-                               <float*> singular_vals_ptr,
-                               <float*> mean_ptr,
-                               params)
+            pcaTransform(h_[0],
+                         <float*> input_ptr,
+                         <float*> components_ptr,
+                         <float*> trans_input_ptr,
+                         <float*> singular_vals_ptr,
+                         <float*> mean_ptr,
+                         params)
         else:
-            pcaTransform(<double*> input_ptr,
-                               <double*> components_ptr,
-                               <double*> trans_input_ptr,
-                               <double*> singular_vals_ptr,
-                               <double*> mean_ptr,
-                               params)
+            pcaTransform(h_[0],
+                         <double*> input_ptr,
+                         <double*> components_ptr,
+                         <double*> trans_input_ptr,
+                         <double*> singular_vals_ptr,
+                         <double*> mean_ptr,
+                         params)
```

Comment:
Line 626 access `trans_input_data` does that need a synchronization before?

Reply:
Same as before. @cjnolet ?

Reply:
same as above comment.